### PR TITLE
feat(CORE/World): Add Rate.MoveSpeedOther to control non-player movespeed as well.

### LINKED ISSUE #18617: Feature: Add Flag to apply Rate.MoveSpeed to NPC and Mobs as well
### Describe your feature request or suggestion in detail

Add a flag such as `Rate.MoveSpeed.ApplyToNonPlayersToo` or another flag such as `Rate.MoveSpeed.Base` to apply to non-players.

This would allow players of the server to move faster while at the same time not breaking the game by allowing you to kite forever as a ranged character.

### Describe a possible solution to your feature or suggestion in detail

In `World.cpp#LoadConfigSettings`, when a a `Rate.MoveSpeed.ApplyToNonPlayersToo` is set, modify the `baseMoveSpeed` as well. Or if another flag for `Rate.MoveSpeed.Base` is set, use that for the base speed. If we go with second option, maybe migrate `Rate.MoveSpeed` -> `Rate.MoveSpeed.Player`.

### AC rev. hash/commit
As of commit: [3ff8de2](https://github.com/azerothcore/azerothcore-wotlk/commit/3ff8de2086a2f32316ec0041963968c6eccf8958)

### PR DESCRIPTION
**Pull request by doowahree**
This change is added to allow faster equal movespeed for players without being able to completely break balance with ranged.

Implements feature #18617

## Changes Proposed:

This PR proposes changes to:
-  [x] Core (units, players, creatures, game systems).
-  [ ] Scripts (bosses, spell scripts, creature scripts).
-  [ ] Database (SAI, creatures, etc).

## Issues Addressed:

- Closes #18617

## SOURCE:

The changes have been validated through:
- [ ] Live research (checked on live servers, e.g Classic WotLK, Retail, etc.)
- [ ] Sniffs (remember to share them with the open source community!)
- [ ] Video evidence, knowledge databases or other public sources (e.g forums, Wowhead, etc.)
- [ ] The changes promoted by this pull request come partially or entirely from another project (cherry-pick). **Cherry-picks must be committed using the proper --author tag in order to be accepted, thus crediting the original authors, unless otherwise unable to be found**

**N/A**: QoL change to keep balance on Rate.MoveSpeed changes.

## Tests Performed:

This PR has been:
- [x] Tested in-game by the author.
- [ ] Tested in-game by other community members/someone else other than the author/has been live on production servers.
- [ ] This pull request requires further testing and may have edge cases to be tested.

Author Tests:
1. Rate.MoveSpeed = 2, Rate.MoveSpeedOther = 1 results in https://youtu.be/jdHkUKVTEBc
2. Rate.MoveSpeed =2, Rate.MoveSpeedOther = 2 results in https://youtu.be/zMllYDAJVUc

## How to Test the Changes:

- [ ] This pull request can be tested by following the reproduction steps provided in the linked issue
- [ ] This pull request requires further testing. Provide steps to test your changes. If it requires any specific setup e.g multiple players please specify it as well.

1. Pull this change.
2. Modify Rate.MoveSpeed to a high number.
3. Test movement speed.
4. Modify Rate.MoveSpeedOther to the same number.
5. Test movement speed.

## Known Issues and TODO List:
None ATM.

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/apps/worldserver/worldserver.conf.dist [hunk 1]
```diff
@@ -2029,11 +2029,18 @@ EnableLowLevelRegenBoost = 1
...
 #    Rate.MoveSpeed
-#        Description: Movement speed rate.
+#        Description: Player movement speed rate.
 #        Default:     1
 
 Rate.MoveSpeed = 1
 
+#
+#    Rate.MoveSpeedOther
+#        Description: Npc and mob movement speed rate.
```
Comment (Kitzunu) [inline src/server/apps/worldserver/worldserver.conf.dist:2039]:
```suggestion
#    Rate.MoveSpeed.NPC
#        Description: NPC movement speed rate.
```

Comment (Kitzunu) [reply src/server/apps/worldserver/worldserver.conf.dist:2039]: change all the necessary spots to make it work too

File: src/server/apps/worldserver/worldserver.conf.dist [hunk 2]
```diff
@@ -2029,11 +2029,18 @@ EnableLowLevelRegenBoost = 1
 
 #
 #    Rate.MoveSpeed
-#        Description: Movement speed rate.
+#        Description: Player movement speed rate.
```
Comment (Kitzunu) [inline src/server/apps/worldserver/worldserver.conf.dist:2032]: I would maybe go as far as changing this to Rate.MoveSpeed.Player

**Comment by sudlud** [comment]
what is the state of this PR, can you please address the reviews

**Comment by sudlud** [comment]
closing this PR as the author does not responded to reviews.

Please feel free to re-open this PR and address the open reviews/comments.

**Pull request closed without merging**